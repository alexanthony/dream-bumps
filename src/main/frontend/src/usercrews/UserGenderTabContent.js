// @flow
import React, {PropTypes, Component} from 'react';
import {Grid, Cell} from 'material-grid/dist';

import UserCrew from './UserCrew';
import UserCurrentScore from './UserCurrentScore';
import UserScoreHistory from './UserScoreHistory';

class UserGenderTabContent extends Component {
  render() {
    return (
      <Grid>
        <Cell col={6}>
          <UserCrew
            userId={this.props.userId}
            header="Current Lineup"
            canBuySell={false}
            sex={this.props.sex}
            showCashAndValue={false}
          />
        </Cell>
        <Cell col={6}>
          <UserCurrentScore
            userId={this.props.userId}
            sex={this.props.sex}
          />
          <UserScoreHistory
            userId={this.props.userId}
            sex={this.props.sex}
            style={{marginTop: '16px'}}
          />
        </Cell>
      </Grid>
    );
  }
}

UserGenderTabContent.propTypes = {
  userId: PropTypes.number,
  sex: PropTypes.string
}

export {UserGenderTabContent};

export default UserGenderTabContent;
