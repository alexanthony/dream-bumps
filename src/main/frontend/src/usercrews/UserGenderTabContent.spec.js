import React from 'react';
import { shallow } from 'enzyme';
import { UserGenderTabContent } from './UserGenderTabContent';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const component = shallow(
    <UserGenderTabContent
      userId={2}
      sex="female"
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
  // We don't actually mount, so these don't get called...
  // expect(fetchUsers).toHaveBeenCalled();
  // expect(fetchUserCrewRankings).toHaveBeenCalled()
});


// UserGenderTabContent.propTypes = {
//   userId: PropTypes.number,
//   sex: PropTypes.string
// }
