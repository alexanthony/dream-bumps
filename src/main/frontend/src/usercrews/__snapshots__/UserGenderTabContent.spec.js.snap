exports[`test Renders properly 1`] = `
<Grid>
  <Cell
    col={6}
    tablet={12}>
    <Connect(UserDetail)
      userId={2} />
    <Connect(UserCrew)
      canBuySell={false}
      header="Current Lineup"
      sex="female"
      showCashAndValue={false}
      style={
        Object {
          "marginTop": "16px",
        }
      }
      userId={2} />
  </Cell>
  <Cell
    col={6}
    tablet={12}>
    <Connect(UserCurrentScore)
      sex="female"
      userId={2} />
    <Connect(UserScoreHistory)
      sex="female"
      style={
        Object {
          "marginTop": "16px",
        }
      }
      userId={2} />
  </Cell>
</Grid>
`;
